File: Solution_Practice_I/java/prob4_3.java
import java.util.Scanner;
public class prob4_3  {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int num,a=1,b=2,c=3,d=4, sum;
        do {
            System.out.println("Ingrese cuantos digitos ver");
            num=input.nextInt();
        } while (num<0);
        for(int i=0; i<num; i++){
            if(i==0) System.out.print(a+" ");
            else if(i==1) System.out.print(b+" ");
            else if(i==2) System.out.print(c+" ");
            else if(i==3) System.out.print(d+" ");
            else{
                sum=a+b+c+d;
                a=b;
                b=c;
                c=d;
                d=sum;
                System.out.print(sum+" ");
            }
        }

        input.close();
    }
}

File: Solution_Practice_I/python/prob4_3.py
while True:
    print("Ingrese cuantos digitos ver:")
    num = int(input())
    if num >= 0:
        break
a=1
b=2
c=3
d=4
for i in range(num):
    if i==0: print(a, end=" ")
    elif i==1: print(b, end=" ")
    elif i==2: print(c, end=" ")
    elif i==3: print(d, end=" ")
    else:
        sum=a+b+c+d
        a=b
        b=c
        c=d
        d=sum
        print(sum, end=" ")